src/components/ObsDetails/ObsDetailsHeader.js

```diff
@@ -0,0 +1,54 @@
+// @flow
+
+import React, { useEffect, useState } from "react";
+import type { Node } from "react";
+import { useTranslation } from "react-i18next";
+import { Headline, Button } from "react-native-paper";
+import { useNavigation } from "@react-navigation/native";
+import { View } from "react-native";
+import { HeaderBackButton } from "@react-navigation/elements";
+
+import { ObsEditContext } from "../../providers/contexts";
+import { viewStyles } from "../../styles/obsDetails/obsDetailsHeader";
+import { colors } from "../../styles/global";
+
+type Props = {
+  observationUUID: string
+}
+
+const ObsDetailsHeader = ( { observationUUID }: Props ): Node => {
+  const [isLocal, setIsLocal] = useState( null );
+  const { openSavedObservation } = React.useContext( ObsEditContext );
+
+  const { t } = useTranslation( );
+  const navigation = useNavigation( );
+
+  const navToObsEdit = ( ) => {
+    navigation.navigate( "camera", {
+      screen: "ObsEdit"
+    } );
+  };
+
+  useEffect( ( ) => {
+    const checkForLocalObservation = async ( ) => {
+      const isLocalObservation = await openSavedObservation( observationUUID );
+      setIsLocal( isLocalObservation );
+    };
+
+    navigation.addListener( "focus", ( ) => {
+      checkForLocalObservation( );
+    } );
+  }, [observationUUID, openSavedObservation, navigation] );
+
+  const goBack = ( ) => navigation.goBack( );
+
+  return (
+    <View style={viewStyles.headerRow}>
+      <HeaderBackButton onPress={goBack} />
+      <Headline>{t( "Observation" )}</Headline>
+      {isLocal ? <Button icon="pencil" onPress={navToObsEdit} color={colors.gray} /> : <View />}
```

Comment:
I'm using the `<View />` here for styling purposes, since the back button, headline, and button are all in a row using `justifyContent: "space-between"`. This ensures that the headline doesn't move to the right side of the screen when there is no Button component. I think once we have final designs it will make more sense to use absolute positioning for each of these header components, I just didn't want to make random guesses on margins & positioning.

Reply:
K. Definitely not a blocker so I'll resolve. FWIW, if we're going to use react-native-paper, we should consider using https://callstack.github.io/react-native-paper/appbar-header.html for all headers like this.